llvm/lib/Target/RISCV/RISCVInstrInfo.td

```diff
@@ -813,6 +815,12 @@ def MRET : Priv<"mret", 0b0011000>, Sched<[]> {
   let rs1 = 0;
   let rs2 = 0b00010;
 }
+
+def MNRET : Priv<"mnret", 0b0111000>, Sched<[]> {
```

Comment:
Does this need a `Requires=[HasStdExtSmrnmi]` (And a definition of `HasStdExtSmrnmi`)?

Reply:
Thanks for your help. I have added the relevant content.

Reply:
Nice, thanks!